src/raftstore/store/fsm/batch.rs

```diff
@@ -243,12 +244,12 @@ pub trait PollHandler<N, C> {
     /// larger than the value. If it returns None, then this function will
     /// still be called for the same FSM in the next loop unless the FSM is
     /// stopped.
-    fn handle_control(&mut self, control: &mut C) -> Option<usize>;
+    fn handle_control(&mut self, control: &mut C, counter: &mut i32) -> Option<usize>;
```

Comment:
Please document `counter`. BTW `handled_counter` maybe is a better name.

Reply:
> it is very strange to pass a counter everywhere.

Please notice my comments. I am so surprised that you LGTM to this change.